lib/coinbase/smart_contract.rb

```diff
@@ -2,257 +2,296 @@
+            }
+          )
+        end
 
-    # Creates a new ERC721 token contract, that can subsequently be deployed to
-    # the blockchain.
-    # @param address_id [String] The address ID of deployer
-    # @param wallet_id [String] The wallet ID of the deployer
-    # @param name [String] The name of the token
-    # @param symbol [String] The symbol of the token
-    # @param base_uri [String] The base URI for the token metadata
-    # @return [SmartContract] The new ERC721 Token SmartContract object
-    def self.create_nft_contract(
-      address_id:,
-      wallet_id:,
-      name:,
-      symbol:,
-      base_uri:
-    )
-      contract = Coinbase.call_api do
-        smart_contracts_api.create_smart_contract(
-          wallet_id,
-          address_id,
-          {
-            type: Coinbase::Client::SmartContractType::ERC721,
-            options: Coinbase::Client::NFTContractOptions.new(
-              name: name,
-              symbol: symbol,
-              base_uri: base_uri
-            ).to_body
-          }
-        )
+        new(contract)
       end
 
-      new(contract)
-    end
+      # Creates a new ERC1155 multi-token contract, that can subsequently be deployed to
+      # the blockchain.
+      # @param address_id [String] The address ID of deployer
+      # @param wallet_id [String] The wallet ID of the deployer
+      # @param uri [String] The URI for the token metadata
+      # @return [SmartContract] The new ERC1155 Multi-Token SmartContract object
+      def create_multi_token_contract(
+        address_id:,
+        wallet_id:,
+        uri:
+      )
+        contract = Coinbase.call_api do
+          smart_contracts_api.create_smart_contract(
+            wallet_id,
+            address_id,
+            {
+              type: Coinbase::Client::SmartContractType::ERC1155,
+              options: Coinbase::Client::MultiTokenContractOptions.new(
+                uri: uri
+              ).to_body
+            }
+          )
+        end
 
-    # Creates a new ERC1155 multi-token contract, that can subsequently be deployed to
-    # the blockchain.
-    # @param address_id [String] The address ID of deployer
-    # @param wallet_id [String] The wallet ID of the deployer
-    # @param uri [String] The URI for the token metadata
-    # @return [SmartContract] The new ERC1155 Multi-Token SmartContract object
-    def self.create_multi_token_contract(
-      address_id:,
-      wallet_id:,
-      uri:
-    )
-      contract = Coinbase.call_api do
-        smart_contracts_api.create_smart_contract(
-          wallet_id,
-          address_id,
-          {
-            type: Coinbase::Client::SmartContractType::ERC1155,
-            options: Coinbase::Client::MultiTokenContractOptions.new(
-              uri: uri
-            ).to_body
-          }
-        )
+        new(contract)
       end
 
-      new(contract)
-    end
+      def register(
+        contract_address:,
+        abi:,
+        name:,
```

Comment:
The contract name is optional, shall we assign a nil by default? `name: nil`

Reply:
We can, but it feels like we should encourage developers to specify a name if possible?

Reply:
Yep make sense